src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -249,28 +265,74 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
         }
 
         // Add a marker to denote the last tab
-        _tabInformation[tabPos].index = 0xFF;
+        _tabPositions[tabPos].index = 0xFF;
 
-        const auto firstTabIndex = ObjectType(_tabInformation[0].index);
+        const auto firstTabIndex = ObjectType(_tabPositions[0].index);
         repositionTargetTab(self, firstTabIndex);
     }
 
-    // 0x00472BBC
-    static ObjectManager::ObjIndexPair sub_472BBC(Window* self)
+    static bool contains(const std::string_view& a, const std::string_view& b)
     {
-        const auto objects = ObjectManager::getAvailableObjects(static_cast<ObjectType>(self->currentTab));
+        return std::search(a.begin(), a.end(), b.begin(), b.end(), [](char a, char b) {
+                   return tolower(a) == tolower(b);
+               })
+            != a.end();
+    }
 
+    static void applyFilterToObjectList()
+    {
+        std::string_view pattern = inputSession.buffer.c_str();
+        _numVisibleObjectsListed = 0;
+        for (auto& entry : _tabObjectList)
+        {
+            if (pattern.empty())
+            {
+                entry.visible = true;
+                _numVisibleObjectsListed++;
+                continue;
+            }
+
+            const std::string_view name = entry.object._name;
+            const std::string_view filename = entry.object._filename;
+
+            const bool containsName = contains(name, pattern);
+            const bool containsFileName = contains(filename, pattern);
+
+            entry.visible = containsName || containsFileName;
+
+            if (entry.visible)
+                _numVisibleObjectsListed++;
+        }
+    }
+
+    static void populateTabObjectList(ObjectType objectType)
+    {
+        _tabObjectList.clear();
```

Comment:
should we clear this on window close? Not sure if its worth it.

Reply:
[`clear`](https://en.cppreference.com/w/cpp/container/vector/clear) doesn't change the capacity so it may not be worth doing. Unless we follow up with [`shrink_to_fit`](https://en.cppreference.com/w/cpp/container/vector/shrink_to_fit)?

Reply:
hmm 28 bytes * 21k objects (that we know exist atm) = 0.5MB. Nah lets just leave it.

Reply:
Yeah, likely less than 1k objects at a time even, and only if the window is even used to begin with. Not really worth it.